src/Processors/QueryPlan/Streaming/ReplayStep.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <Processors/QueryPlan/ITransformingStep.h>
+#include <Common/IntervalTree.h>
+#include <Processors/QueryPlan/IQueryPlanStep.h>
+#include <base/types.h>
+
+namespace DB
+{
+namespace Streaming
+{
+class ReplayStep : public ITransformingStep
```

Comment:
1. It will not be inherited, so you need to add the modifier `final`
2. Rename it to `ReplayStreamStep`
such as `class ReplayStreamStep final : public ITransformingStep`